packages/services/api/src/modules/integrations/providers/github-integration-manager.ts

```diff
@@ -177,18 +179,42 @@ export class GitHubIntegrationManager {
 
     const octokit = await this.app.getInstallationOctokit(parseInt(installationId, 10));
 
-    const result = await octokit.request('POST /repos/{owner}/{repo}/check-runs', {
-      owner: input.repositoryOwner,
-      repo: input.repositoryName,
-      name: input.name,
-      head_sha: input.sha,
-      conclusion: input.conclusion,
-      output: input.output,
-      details_url: input.detailsUrl ?? undefined,
-    });
+    try {
+      const result = await octokit.request('POST /repos/{owner}/{repo}/check-runs', {
+        owner: input.repositoryOwner,
+        repo: input.repositoryName,
+        name: input.name,
+        head_sha: input.sha,
+        conclusion: input.conclusion,
+        output: input.output,
+        details_url: input.detailsUrl ?? undefined,
+      });
+
+      this.logger.debug('Check-run created (link=%s)', result.data.url);
 
-    this.logger.debug('Check-run created (link=%s)', result.data.url);
+      return result.data.url;
+    } catch (error) {
+      this.logger.error('Failed to create check-run', error);
 
-    return result.data.url;
+      if (isOctokitRequestError(error)) {
+        this.logger.debug(
+          'GitHub error details (message=%s, status=%s)',
+          error.message,
+          error.status,
+        );
+
+        if (error.message.includes('No commit found for SHA:')) {
+          throw new HiveError(
+            `Commit ${input.sha} not found in repository ${input.repositoryOwner}/${input.repositoryName}`,
+          );
+        }
```

Comment:
Instead of throwing `No commit found for SHA: 123456`, we add more details (useful to detect a wrong repo connected to a project)